icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5323,4 +5324,27 @@ void RBBITest::TestTraceCreateBreakEngine(void) {
 }
 #endif
 
+void RBBITest::TestUnpairedSurrogate() {
+    UnicodeString rules(u"ab;");
+
+    UErrorCode status = U_ZERO_ERROR;
+    UParseError pe;
+    RuleBasedBreakIterator bi1(rules, pe, status);
+    assertSuccess(WHERE, status);
+    UnicodeString rtRules = bi1.getRules();
+    // make sure the simple one work first.
+    assertEquals(WHERE, UnicodeString(u"ab;"),  rtRules);
+
+    char16_t r2[] = {'a', 0xd800, ';', 0};
+    UnicodeString rules2(r2);
+    RuleBasedBreakIterator bi2(rules2, pe, status);
+    assertEquals("unpaired lead surrogate", U_INVALID_CHAR_FOUND , status);
+
```

Comment:
Need to check that the UParseError is set correctly. If the check is moved into rbbiscan.cpp, it should be.

Reply:
ping please check the contents of `pe` in the error cases